File: couponkill-order-service/src/main/java/com/aliyun/seckill/order/service/OrderService.java
// com.aliyun.seckill.order.service.OrderService.java
package com.aliyun.seckill.order.service;

import com.aliyun.seckill.pojo.Order;
import com.baomidou.mybatisplus.extension.service.IService;

public interface OrderService extends IService<Order> {
    /**
     * 创建订单(领取优惠券)
     */
    Order createOrder(Long userId, Long couponId);

    /**
     * 取消订单(退还优惠券)
     */
    boolean cancelOrder(String orderId, Long userId);

    /**
     * 检查用户是否已领取该优惠券
     */
    boolean hasUserReceivedCoupon(Long userId, Long couponId);

    /**
     * 检查用户优惠券数量是否超过限制
     */
    boolean checkCouponCountLimit(Long userId, int couponType);
}

File: frontend/couponkill-frontend/src/services/orderService.ts
import { http } from '../lib/apiClient'
import type { Order } from '../types/api'

// Phase1：统一走 /api/v1/order/**（网关与后端仍兼容旧 /order/**）
// 身份由网关 JWT → X-User-Id 注入，不再传可伪造的 userId
const ORDER_BASE = '/api/v1/order'

export const orderService = {
  async getUserOrders(pageNum = 1, pageSize = 10): Promise<Order[]> {
    return http.get<Order[]>(`${ORDER_BASE}/user/me`, {
      params: { pageNum, pageSize },
    })
  },

  async createOrder(couponId: string): Promise<Order> {
    return http.post<Order>(`${ORDER_BASE}/create`, null, {
      params: { couponId },
    })
  },

  async seckill(couponId: string): Promise<unknown> {
    return http.post<unknown>(`${ORDER_BASE}/seckill`, null, {
      params: { couponId },
    })
  },

  async cancelOrder(orderId: string): Promise<boolean> {
    return http.post<boolean>(`${ORDER_BASE}/cancel`, null, {
      params: { orderId },
    })
  },

  async checkReceived(couponId: string): Promise<boolean> {
    return http.get<boolean>(`${ORDER_BASE}/check/${couponId}`)
  },
}

export default orderService
